xtask/codegen/js.ungram

```diff
@@ -2125,14 +2125,18 @@ TsMappedTypeReadonlyModifierClause =
 	operator_token: ('+' | '-')?
 	'readonly'
 
+TsImportArguments = '(' args: TsImportArgumentList ')'
+
+TsImportArgumentList = (AnyJsCallArgument (',' AnyJsCallArgument)* )
+
 // a: import("./test").T<typeof X>
 //    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 // a: import("./test")<string>
 //    ^^^^^^^^^^^^^^^^^^^^^^^^^
 TsImportType =
 	'typeof'?
 	'import'
-	arguments: JsCallArguments
+	arguments: TsImportArguments
```

Comment:
Hey, based on the typescript source, it looks like it should be like this:

```
// (not tested)
TsImportType =
	'typeof'?
	'import'
	argument: AnyTsType
	(
		','
		'{'
		('assert' | 'with')
		':'
		'{'
		assertions: TsImportTypeAssertionList
		'}'
		'}'
	)?
	qualifier_clause: TsImportTypeQualifier?
	type_arguments: TsTypeArguments?

TsImportTypeAssertionList =
	(TsImportTypeAssertion (',' TsImportTypeAssertion)* ','?)

TsImportTypeAssertion =
	key: ('identifier' | 'js_string_literal')
	':'
	value: AnyJsExpression // parse_assignment_expression_or_higher
```

https://github.com/microsoft/TypeScript/blob/main/src/compiler/parser.ts#L4546

 * The first argument is any TS type
     * (not only a string - the "String literal expected" error comes from the typechecker)
 * The `,{assert:{` and `}}` are hardcoded tokens, it's not a comma separated list
 * The part inside the inner braces is a comma separated list of `key:expression` where key can only be an identifier or string literal, and expression is a JS expression (parseAssignmentExpressionOrHigher).
     * Trailing commas are allowed

Examples:
```
type A = typeof import(1); // ok (typechecker error)
type B = typeof import("a.json",{}); // syntax error - missing assert keyword
type C = typeof import("a.json",{assert:{}}); // ok
type D = typeof import("a.json",{assert:{a:"1"}}); // ok
type E = typeof import("a.json",{assert:{[a]:"1"}}); // syntax error - keys need to be identifiers or string literals
type F = typeof import("a.json",{assert:{"a":"1"}}); // ok
type G = typeof import("a.json",{assert:{a:"1", b:"2",}}); // ok
type H = typeof import("a.json",{assert:{}}, 1); // syntax error - 3rd arg not valid
type I = typeof import(import(Record<string,string>),{assert:{a:await p}}); // ok (typechecker error)
```
https://www.typescriptlang.org/play/?#code/CYUwxgNghgTiAEBbA9sArhBAiKA6AVgM7IB2W8A3gL4BQNALgJ4AOCAgvALzxOvIBm8AJaJmyGPQAUARgCUAbngB6JfGQBreJN7gAFuHUgY8IzHGyGLBACEuPKwOGjxUnAWJkANNQXLVhRhJ6KAAPExgzYwBaJCFCQiESAHN4KHijenhDRgB3cWBLVngAYTsdRxExCUk3IlIsbzTCDIAuaipfFTV1QoQAETKHQUqXGrw6rwom1qmWrGksKg7FLo1e+ABRQb5h52rajwap9Ik2gG0oAF05haXO-0DgsNNxeBjswngSEBBgHmR4AAjBBCUBBIT8IRGT6vQj0GCJFIQIT0IxQCCEdYAMW2IAqe1c40OjRO9DaOCwN0Wyz83XWAHFcfiqoT3PUSc1TrN5g0gXMAEwNO4rVRrHTwAASTN2LLGbMm0y5S088DkIvgASCoXCkTe8AAzDA-rAUiRkJkAG7o0HrACS0qcspG1QASuB8gAeOEI5Keb2IgB8sg5MygLSgOSgKPgzGFtI0Wh0YH0YEMxheMFkQA

Reply:
😯, thank you very much to provide the typescript context, i will base this on to update my code here, I haven't seen the relevant parser specification reference for this before.